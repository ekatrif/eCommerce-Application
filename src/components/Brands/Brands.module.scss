.list {
  display: flex;
  gap: 2em;
  justify-content: center;
  margin: 2em 0;
}

.item {
  opacity: 0.4;
  filter: grayscale(100%);
  transition: all 0.5s linear;

  &:hover {
    opacity: 1;
    filter: grayscale(0%);
    transition: all 0.5s linear;
  }
}