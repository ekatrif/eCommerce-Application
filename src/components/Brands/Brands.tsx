import React from 'react';
import { Link } from 'react-router-dom';
import classes from './Brands.module.scss';
import brands from '../../assets/data/main-brands';

const Brands: React.FC = () => {
  return (
    <ul className={classes.list}>
      {brands.length
        ? brands.map((brand) => (
            <li className={classes.item} key={brand.id}>
              <Link to="/catalog">
                <img src={brand.url} alt={brand.alt} />
              </Link>
            </li>
          ))
        : null}
    </ul>
  );
};

export default Brands;
