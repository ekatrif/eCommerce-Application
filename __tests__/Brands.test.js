import React from 'react';
import renderer from 'react-test-renderer';
import Brands from '../src/components/Brands/Brands';
import { BrowserRouter } from 'react-router-dom'

test('Brands renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><Brands /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
