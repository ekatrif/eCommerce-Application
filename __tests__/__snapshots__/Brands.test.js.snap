// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands renders correctly 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <a
      href="/catalog"
      onClick={[Function]}
    >
      <img
        alt="Adidas"
        src="/assets/images/brands-main/adidas.jpg"
      />
    </a>
  </li>
  <li
    className="item"
  >
    <a
      href="/catalog"
      onClick={[Function]}
    >
      <img
        alt="Columbia"
        src="/assets/images/brands-main/columbia.jpg"
      />
    </a>
  </li>
  <li
    className="item"
  >
    <a
      href="/catalog"
      onClick={[Function]}
    >
      <img
        alt="Mammut"
        src="/assets/images/brands-main/mammut.jpg"
      />
    </a>
  </li>
  <li
    className="item"
  >
    <a
      href="/catalog"
      onClick={[Function]}
    >
      <img
        alt="Salewa"
        src="/assets/images/brands-main/salewa.jpg"
      />
    </a>
  </li>
  <li
    className="item"
  >
    <a
      href="/catalog"
      onClick={[Function]}
    >
      <img
        alt="Solomon"
        src="/assets/images/brands-main/solomon.jpg"
      />
    </a>
  </li>
</ul>
`;
